validations/src/validations.rs

```diff
@@ -161,24 +200,21 @@ pub fn validate_dr_transaction(tx: &TransactionBody) -> Result<(), failure::Erro
 
 // Add 1 in the number assigned to a DataRequestOutput
 pub fn update_count<S: ::std::hash::BuildHasher>(
-    mut hm: HashMap<DataRequestOutput, u32, S>,
-    k: &DataRequestOutput,
+    hm: &mut WitnessesCounter<S>,
+    k: &OutputPointer,
+    rf: u32,
 ) {
-    match hm.get_mut(k) {
-        Some(count) => {
-            *count += 1;
-        }
-        None => {
-            hm.insert(k.clone(), 1);
-        }
-    };
+    hm.entry(k.clone()).or_insert((0, rf)).0 += 1;
 }
 
+/// HashMap to count commit and reveals transactions need for a Data Request
+pub type WitnessesCounter<S> = HashMap<OutputPointer, (u32, u32), S>;
```

Comment:
_Suggestion_: describe what represents each field of the tuple, or use a struct
```rust
pub struct WitnessesCount {
  commits: u32,
  reveals: u32
}
pub type WitnessesCounter<S> = HashMap<OutputPointer, WitnessesCount, S>;
```

Also, why do you parameterize with `S`?

Reply:
Clippy likes the S

Reply:
For reference: https://doc.rust-lang.org/src/std/collections/hash/map.rs.html#409-416